File: async-scala-client/src/main/scala/io/swagger/client/model/CalculatedTaxSummaryForGoods.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class CalculatedTaxSummaryForGoods (
  numberOfLines: Option[Integer],  // Count of lines
taxedDiscount: Option[Double],  // sum of all line taxed discounts
untaxedDiscount: Option[Double],  // sum of all line untaxed discounts
subtotal: Option[Double],  // sum of all line lineAmount attribute
totalTax: Option[Double],  // sum of all line tax attribute
grandTotal: Option[Double],  // sum of all line lineAmount attribute + sum of all line tax attribute not VAT - sum of all line lineTaxedDiscount attribute
taxByType: Option[CalculatedTaxSummaryForGoods_taxByType]
)

File: javascript-closure-angular-client/API/Client/CalculatedTaxSummaryForGoods.js
goog.provide('API.Client.CalculatedTaxSummaryForGoods');

/**
 * @record
 */
API.Client.CalculatedTaxSummaryForGoods = function() {}

/**
 * Count of lines
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.numberOfLines;

/**
 * sum of all line taxed discounts
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.taxedDiscount;

/**
 * sum of all line untaxed discounts
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.untaxedDiscount;

/**
 * sum of all line lineAmount attribute
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.subtotal;

/**
 * sum of all line tax attribute
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.totalTax;

/**
 * sum of all line lineAmount attribute + sum of all line tax attribute not VAT - sum of all line lineTaxedDiscount attribute
 * @type {!number}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.grandTotal;

/**
 * @type {!API.Client.CalculatedTaxSummaryForGoods_taxByType}
 * @export
 */
API.Client.CalculatedTaxSummaryForGoods.prototype.taxByType;

